### PR TITLE
Bulk retry on namestale

### PR DESCRIPTION
**Pull request by ealsur**
# Pull Request Template

## Description

Bulk flow should retry on NameStale Cache, this PR modifies the retry policy to cover that case and adds a UT to verify it.

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/BulkPartitionKeyRangeGoneRetryPolicy.cs [hunk 1]
```diff
@@ -83,7 +83,7 @@ public void OnBeforeSendRequest(DocumentServiceRequest request)
             string resourceIdOrFullName)
         {
             if (statusCode == HttpStatusCode.Gone
-                && subStatusCode == SubStatusCodes.PartitionKeyRangeGone
+                && (subStatusCode == SubStatusCodes.PartitionKeyRangeGone || subStatusCode == SubStatusCodes.NameCacheIsStale)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/BulkPartitionKeyRangeGoneRetryPolicy.cs]: PartitionKeyRangeGone is generic and super set of NameCache stale. 
Just UT suffice right?

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/BulkPartitionKeyRangeGoneRetryPolicy.cs]: image

image

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/BulkPartitionKeyRangeGoneRetryPolicy.cs]: NameStale is 1000, PKRangeGone is 1002, they are not equal

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/BulkPartitionKeyRangeGoneRetryPolicy.cs]: Apologies.

**Pull request merged**